internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -728,6 +728,189 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 				},
 			},
 
+			"maintenance_window_auto_upgrade": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"frequency": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"Weekly",
+								"RelativeMonthly",
+								"AbsoluteMonthly",
+							}, false),
+						},
+
+						"interval": {
+							Type:     pluginsdk.TypeInt,
+							Required: true,
+						},
+
+						"day_of_week": {
+							Type:     pluginsdk.TypeString,
+							Optional: true,
+							ValidateFunc: validation.StringInSlice(
+								maintenanceconfigurations.PossibleValuesForWeekDay(),
+								false),
+						},
+
+						"duration": {
+							Type:         pluginsdk.TypeInt,
+							Required:     true,
+							ValidateFunc: validation.IntBetween(4, 24),
+						},
+
+						"week_index": {
+							Type:     pluginsdk.TypeString,
+							Optional: true,
+							ValidateFunc: validation.StringInSlice(
+								maintenanceconfigurations.PossibleValuesForType(),
+								false),
+						},
+
+						"day_of_month": {
+							Type:         pluginsdk.TypeInt,
+							Optional:     true,
+							ValidateFunc: validation.IntBetween(0, 31),
+						},
+
+						"start_date": {
+							Type:     pluginsdk.TypeString,
+							Optional: true,
+							Computed: true,
+						},
+
+						"start_time": {
+							Type:     pluginsdk.TypeString,
+							Optional: true,
+						},
+
+						"utc_offset": {
+							Type:     pluginsdk.TypeString,
+							Optional: true,
+						},
+
+						"not_allowed": {
+							Type:     pluginsdk.TypeSet,
+							Optional: true,
+							Elem: &pluginsdk.Resource{
+								Schema: map[string]*pluginsdk.Schema{
+									"end": {
+										Type:             pluginsdk.TypeString,
+										Required:         true,
+										DiffSuppressFunc: suppress.RFC3339Time,
+										ValidateFunc:     validation.IsRFC3339Time,
+									},
+
+									"start": {
+										Type:             pluginsdk.TypeString,
+										Required:         true,
+										DiffSuppressFunc: suppress.RFC3339Time,
+										ValidateFunc:     validation.IsRFC3339Time,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"maintenance_window_node_os": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"frequency": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"Weekly",
+								"RelativeMonthly",
+								"AbsoluteMonthly",
+								"Daily",
```

Comment:
Same here

Reply:
Same answer here as well, these are introduced enums as we have currently a flatter structure than the API. We can circumvent it by creating blocks

```hcl
...
    daily {
      interval = 1
    }
    weekly {
      interval    = 1
      day_of_week = "Monday"
    }
    relative_monthly {
      interval    = 1    
      day_of_week = "Monday"
      week_index  = "First"
    }
    absolute_monthly {
      interval     = 1
      day_of_month = 5
    }
...
```